index.js

```diff
@@ -101,7 +99,7 @@ function equal(a, b) {
       }
 
       // all other properties should be traversed as usual
-      if (!equal(a[key], b[key])) return false;
+      if (!equal(a[keys[i]], b[keys[i]])) return false;
```

Comment:
```suggestion
      if (
        (keys[i] !== '_owner' || !a.$$typeof) &&
          !equal(a[keys[i]], b[keys[i]])
      ) {
        return false;
      }
```

Combining this check with the React-specific one above is safe.

Reply:
Yeah, this one works logic-wise! Unfortunately, I applied this diff:

```diff
diff --git a/index.js b/index.js
index cafa587..3635858 100644
--- a/index.js
+++ b/index.js
@@ -89,18 +89,12 @@ function equal(a, b) {
     if (hasElementType && a instanceof Element) return false;
 
     // custom handling for React
-    for (i = length; i-- !== 0;) {
-      if (keys[i] === '_owner' && a.$$typeof) {
-        // React-specific: avoid traversing React elements' _owner.
-        //  _owner contains circular references
-        // and is not needed when comparing the actual elements (and not their owners)
-        // .$$typeof and ._store on just reasonable markers of a react element
-        continue;
-      }
-
-      // all other properties should be traversed as usual
-      if (!equal(a[keys[i]], b[keys[i]])) return false;
-    }
+    for (i = length; i-- !== 0;)
+      // React-specific: avoid traversing React elements' _owner.
+      //  _owner contains circular references
+      // and is not needed when comparing the actual elements (and not their owners)
+      // .$$typeof and ._store on just reasonable markers of a react element
+      if ((keys[i] !== '_owner' || !a.$$typeof) && !equal(a[keys[i]], b[keys[i]])) return false;
     // END: react-fast-compare
 
     // START: fast-deep-equal
```

and `yarn size-min-gz` remained the same at `651` so no change in min+gz bytesize.